File: grider/reverseint/reverseint.go
package reverseint

import (
	"strconv"
)

func ReverseInt(n int) int {
	tmpN := n
	if n < 0 {
		tmpN = tmpN * -1
	}
	stringTmpN := ""

	for tmpN > 0 {
		remain := tmpN % 10
		tmpN = tmpN / 10

		stringRemain := strconv.Itoa(remain)
		stringTmpN = stringTmpN + stringRemain
	}

	newN, _ := strconv.Atoi(stringTmpN)

	if n < 0 {
		newN = newN * -1
	}

	return newN
}

func ReverseInt2(n int) int {
	tmpN := n
	if n < 0 {
		tmpN = tmpN * -1
	}

	reversedN := 0

	for tmpN > 0 {
		remainder := tmpN % 10
		tmpN = tmpN / 10

		reversedN = (reversedN * 10) + remainder
	}

	if n < 0 {
		reversedN = reversedN * -1
	}

	return reversedN
}

File: grider/reverseint/reverseint.py
# class Solution:
#     def reverseint(num):
#         return 0

def reverseint(num):
    tmpNum = num
    if tmpNum < 0:
        tmpNum *= -1

    n = 0
    while(tmpNum > 0):
        remainder = tmpNum % 10
        tmpNum = tmpNum // 10
        n = (n * 10) + remainder

    if num < 0:
        n *= -1

    return n